### PR TITLE
Instantiate DualSpace template in bindings config file

### PR DESCRIPTION
**Pull request by nunoguedelha**
This change fixes issue https://github.com/robotology/idyntree/issues/105. The dot product between Spatial Vectors was broken in the bindings.

Some context... The respective function is defined in the class `SpatialVector` as :
```
typedef typename DualSpace<DerivedSpatialVecT>::Type DualVectorT;
...
double dot(const DualVectorT & other) const;
```
where `DualSpace ` is the template defining the dual-space vector class with respect to the parent class of the **dot** operator such that: the dot product invoked by a `SpatialMotionVector` will take as input a `SpatialForceVector ` and vice-versa.

The `dot` function prototype cannot be defined without instantiating the `DualSpace ` template, but, as it has been observed previously, SWIG is unable to handle on his own the template instantiations. For that reason we have to specify all template instantiations in iDynTree.i config file.

### CODE REVIEW

**Comment by nunoguedelha** [comment]
`_iDynTree.so` wouldn't be found by the Python script if we build iDynTree using Xcode. This was fixed here by https://github.com/robotology/idyntree/pull/201/commits/e925d37f38515c89788e7140513f653409279b35.

File: bindings/matlab/iDynTreeMATLAB_wrap.cxx [hunk 1]
```diff
@@ -1,6 +1,6 @@
 /* ----------------------------------------------------------------------------
  * This file was automatically generated by SWIG (http://www.swig.org).
- * Version 3.0.9
+ * Version 3.0.5
```
Comment (traversaro) [inline bindings/matlab/iDynTreeMATLAB_wrap.cxx]: With which commit did you generate the bindings? https://github.com/jaeandersson/swig/tree/5d3770e452b71265162001c2249dea7961660629 should be 3.0.9 .

Comment (nunoguedelha) [inline bindings/matlab/iDynTreeMATLAB_wrap.cxx]: Sorry, I hadn't compiled properly SWIG after selecting the correct commit. Now it's SWIG build has been done, bindings regenerated and pushed again.

**Comment by traversaro** [comment]
Merging as the OS X failures are not just random failures of travis , not directly related to the PR (I hope).

**Pull request merged**